src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -605,97 +612,135 @@ public HttpResponse onSuccess(String authorizationCode) {
         };
     }
     
-    public void updateAuthorizationStrategy(OAuthSession oauth) throws IOException, GeneralSecurityException {
+    public UsernamePasswordAuthenticationToken updateAuthorizationStrategy(Credential credential) throws IOException, GeneralSecurityException {
     	populateDefaults();
-        OpenShiftUserInfo info = getOpenShiftUserInfo(oauth.getCredential(), transport);
-        Set<String> allowedRoles = postSAR(oauth.getCredential(), transport);
+        OpenShiftUserInfo info = getOpenShiftUserInfo(credential, transport);
+        Set<String> allowedRoles = postSAR(credential, transport);
         GrantedAuthority[] authorities = new GrantedAuthority[] {SecurityRealm.AUTHENTICATED_AUTHORITY};
-        // logs this user in.
-        UsernamePasswordAuthenticationToken token =
-                new UsernamePasswordAuthenticationToken(info.getName(), "", authorities);
-        SecurityContextHolder.getContext().setAuthentication(token);
-        
-        // So if you look at GlobalSecurityConfiguration and GlobalMatrixAuthorizationStrategy (including its DescriptorImpl)
-        // and the associated config.jelly files,
-        // you'll see that the AuthourizationStrategy object stored in Jenkins is *essentially* immutable (except for adds, with comments saying only to use durin contruction), 
-        // and that when users manipulate 
-        // the panel "Configure Global Security", new instances of Global/ProjectMatrixAuthorizationStrategy are created, where
-        // existing users are set up again.
-        // we'll mimic what the "Configure Global Security" config page does                                        
+                
+        // we append the role suffix to the name stored into Jenkins, since a given user is able to log in at varying scope/permission
+        // levels in openshift; however, for now, we make sure the display name for Jenkins does not include this suffix
+        String suffix = "";
+        if (allowedRoles.contains("admin"))
+            suffix = "-admin";
+        else if (allowedRoles.contains("edit"))
+            suffix = "-edit";
+        else if (allowedRoles.contains("view"))
+            suffix = "view";
```

Comment:
-view is consistent with the others (though technically it works with just "view" ... i'll update and push